### PR TITLE
AP_Motors: add pre-arm

### PR DESCRIPTION
**Pull request by IamPete1**
Moves motors pre arms to AP Motors. Also adds a arming check for that all expected motors are setup.

### CODE REVIEW

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -44,7 +44,9 @@ bool AP_Arming_Copter::run_pre_arm_checks(bool display_failure)
     }
 
     // always check motors
-    if (!motor_checks(display_failure)) {
+    char failure_msg[50] = {};
```
Comment (muramura) [inline ArduCopter/AP_Arming.cpp:47]: The number of character bytes in a MAVLINK message is 50 bytes. Therefore, a string that does not exceed 50 bytes is good.

File: ArduCopter/AP_Arming.cpp [hunk 2]
```diff
@@ -44,7 +44,9 @@ bool AP_Arming_Copter::run_pre_arm_checks(bool display_failure)
...
+    if (!copter.motors->arming_checks(ARRAY_SIZE(failure_msg), failure_msg)) {
+        check_failed(display_failure, "Motors: %s", failure_msg);
```
Comment (muramura) [inline ArduCopter/AP_Arming.cpp:49]: The number of character bytes in a MAVLINK message is 50 bytes. Therefore, a string that does not exceed 50 bytes is good.

File: libraries/AP_Motors/AP_MotorsHeli.cpp [hunk 3]
```diff
@@ -603,3 +603,17 @@ void AP_MotorsHeli::update_turbine_start()
     }
 }
 
+bool AP_MotorsHeli::arming_checks(size_t buflen, char *buffer) const
+{
+    // run base class checks
+    if (!AP_Motors::arming_checks(buflen, buffer)) {
+        return false;
+    }
+
```
Comment (bnsgeyer) [inline libraries/AP_Motors/AP_MotorsHeli.cpp:612]: Probably want to run a similar check that multi runs in it's arming check function

File: ArduCopter/AP_Arming.cpp (hunk 1 above)
Comment (peterbarker) [inline ArduCopter/AP_Arming.cpp:47]:
```suggestion
    char failure_msg[50] {};
```

File: ArduPlane/AP_Arming.cpp [hunk 4]
```diff
@@ -142,8 +142,9 @@ bool AP_Arming_Plane::quadplane_checks(bool display_failure)
         ret = false;
     }
 
-    if (!plane.quadplane.motors->initialised_ok()) {
-        check_failed(display_failure, "Quadplane: check motor setup");
+    char failure_msg[50] = {};
```
Comment (peterbarker) [inline ArduPlane/AP_Arming.cpp:145]:
```suggestion
    char failure_msg[50] {};
```

**Review by rmackay9 — APPROVED** [review]
Looks ok to me.

**Comment by IamPete1** [comment]
Tested on tricopter and tailstter, not tested on coax or single copter but I'm sure it would have done the wrong thing so they no longer run the motor allocation check.

**Comment by rmackay9** [comment]
I think we are just waiting for @bnsgeyer to say he's OK with this right?

File: ArduCopter/AP_Arming.cpp [hunk 5]
```diff
@@ -44,7 +44,9 @@ bool AP_Arming_Copter::run_pre_arm_checks(bool display_failure)
     }
 
     // always check motors
-    if (!motor_checks(display_failure)) {
+    char failure_msg[50] {};
+    if (!copter.motors->arming_checks(ARRAY_SIZE(failure_msg), failure_msg)) {
+        check_failed(display_failure, "Motors: %s", failure_msg);
```
Comment (muramura) [inline ArduCopter/AP_Arming.cpp:49]: I only had the message notification sent in two parts.
This requires that all messages be put into 50 bytes.
The messages are sent in STATUSTEXT messages.
https://mavlink.io/en/messages/common.html#STATUSTEXT

Comment (rmackay9) [reply ArduCopter/AP_Arming.cpp:49]: I agree that we should try and keep messages to display on a single line if possible.  We didn't consider this a blocker but maybe worth a follow-up PR but we will leave this to @IamPete1 to decide.

**Pull request merged**